python/cudf_polars/cudf_polars/callback.py

```diff
@@ -52,19 +148,29 @@ def execute_with_cudf(
     nt
         NodeTraverser
 
-    raise_on_fail
-        Should conversion raise an exception rather than continuing
-        without setting a callback.
+    config
+        GPUEngine configuration object
 
     exception
         Optional exception, or tuple of exceptions, to catch during
         translation. Defaults to ``Exception``.
 
     The NodeTraverser is mutated if the libcudf executor can handle the plan.
     """
+    device = config.device
```

Comment:
We should also validate these right?

Reply:
I suppose it's less bad since we get a TypeError
(from rmm Cython stuff I think)

Reply:
Everything is type-annotated on the polars side, so I am inclined to let this slide, unless it is the case that passing something invalid "succeeds".

Reply:
Added tests that these do raise for invalid values.